hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieCreateHandle.java

```diff
@@ -139,23 +140,34 @@ protected void doWrite(HoodieRecord record, Schema schema, TypedProperties props
           return;
         }
 
-        MetadataValues metadataValues = new MetadataValues().setFileName(path.getName());
-        HoodieRecord populatedRecord =
-            record.prependMetaFields(schema, writeSchemaWithMetaFields, metadataValues, config.getProps());
-
-        if (preserveMetadata) {
-          fileWriter.write(record.getRecordKey(), populatedRecord, writeSchemaWithMetaFields);
-        } else {
-          fileWriter.writeWithMetadata(record.getKey(), populatedRecord, writeSchemaWithMetaFields);
+        // Inject custom logic for the record.
+        Option<Pair<HoodieRecord, Schema>> processedRecord = config.getRecordMerger().insert(record, schema, config.getProps());
+        if (!processedRecord.isPresent()
+            || HoodieOperation.isDelete(processedRecord.get().getLeft().getOperation())
```

Comment:
We can revert all the changes to create handle, the delete messages already got handled in line 137.

Reply:
Should we allow the custom logic?